flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPool.java

```diff
@@ -186,6 +203,22 @@ public void recycle(MemorySegment segment) {
 				if (segment != null) {
 					segments.add(segment);
 				}
+
+				if (segments.size() >= numberOfSegmentsToRequest) {
+					break;
+				}
+
+				if (!deadline.hasTimeLeft()) {
+					throw new IOException(String.format("Insufficient number of network buffers: " +
```

Comment:
This exception throwing looks duplicated. If the message is indeed identical everywhere, can you extract it to `throwInsufficientNumberOfNetworkBuffers()` method?

or maybe we extract the construction of the 
> The total number of network buffers is currently set to %d of %d bytes each. You can increase this number by setting the configuration keys '%s', '%s', and '%s'."

message to a common method, and here throw a slightly modified exception with a mention that "timeout was triggered, and you also might want to increase the timeout value"?

Reply:
Very thanks for the comments, I preferred to the second method since the timeout should need a different exception message. I have extracted the construction and changed the exception message for requesting timeout.